src/tests/ftest/object/same_key_different_value.py

```diff
@@ -176,63 +159,57 @@ def test_array_to_single_value(self):
                 read_back_data = self.container.read_an_array(
                     len(array_value_data), length + 1, dkey, akey, obj)
 
                 for j in range(3):
-                    if (array_value_data[j][0:length-1] !=
-                            read_back_data[j][0:length-1]):
+                    if (array_value_data[j][0:length - 1] != read_back_data[j][0:length - 1]):
                         self.log.info("Written Data: %s", array_value_data[j])
                         self.log.info("Read Data:    %s", read_back_data[j])
-                        self.fail("Data mismatch\n")
+                        self.fail("Data mismatch")
                 # test case 1
                 if i == 0:
                     try:
                         # write single value data to same keys, expected to fail
-                        self.container.write_an_obj(single_value_data,
-                                                    len(single_value_data)+1,
-                                                    dkey, akey, obj, obj_cls=1)
+                        self.container.write_an_obj(
+                            single_value_data, len(single_value_data) + 1,
+                            dkey, akey, obj, obj_cls=1)
                         # above line is expected to return an error,
                         # if not fail the test
                         self.fail("Single value write to existing array value"
                                   + " key should have failed\n")
 
                     # should fail with -1001 ERR
-                    except DaosApiError as excp:
-                        if "-1001" not in str(excp):
-                            self.log.info(excp)
-                            self.fail("Should have failed with -1001 error"
-                                      + " message, but it did not\n")
+                    except DaosApiError as error:
+                        if "-1001" not in str(error):
+                            self.log.error(error)
+                            self.fail("Expected write to fail with -1001")
 
                 # test case 2 and 3
-                elif i == 1 or 2:
+                else:
                     try:
                         # punch the keys
                         obj.punch_akeys(0, dkey, [akey])
                         obj.punch_dkeys(0, [dkey])
 
-                        if aggregation is True:
+                        if aggregation:
                             # trigger aggregation
                             self.container.aggregate(self.container.coh, 0)
 
                         # write to the same set of keys under same object
                         # with single value type
-                        self.container.write_an_obj(single_value_data,
-                                                    len(single_value_data)+1,
-                                                    dkey, akey, obj, obj_cls=1)
+                        self.container.write_an_obj(
```

Comment:
(pylint-no-member) Instance of 'TestContainer' has no 'write_an_obj' member
(pylint-no-member) Instance of 'list' has no 'write_an_obj' member